lib/class-controller.php

```diff
@@ -130,6 +130,37 @@ public function enable_relative_date_aggregation( array $args = [] ): Controller
 		return $this;
 	}
 
+	/**
+	 * Enables search-as-you-type suggestions.
+	 *
+	 * @param array $args {
+	 *     Optional. An array of arguments.
+	 *
+	 *     @type string[] $post_types   Limit suggestions to this subset of all
+	 *                                  indexed post types.
+	 *     @type bool     $show_in_rest Whether to register REST API search handlers
+	 *                                  for querying suggestions. Default true.
+	 * }
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_search_suggestions( array $args = [] ): Controller {
+		$args = wp_parse_args(
+			$args,
+			[
+				'post_types'   => [],
+				'show_in_rest' => true,
+			]
+		);
+
+		$args['post_types'] = array_filter( (array) $args['post_types'] );
```

Comment:
Do we need to confirm if the submitted post types are valid? Considering one can add any `string`.

Reply:
That type of validation doesn't occur while registering other features, like taxonomy aggregations, so, for consistency, I'm hesitant to add it. Plus, `elasticsearch_extensions_config` fires at `init:1000`, and if someone wants to register their post type on `init:1001`, why create a hassle for them?

Your suggestion did make me pause, though, and I think it does make sense to add that validation before running the direct ES query for suggestions to avoid exposing any stale content that might be lingering in the index.

Reply:
> That type of validation doesn't occur while registering other features, like taxonomy aggregations,

I noticed that as well. Maybe we should move this conversation outside of this pull request. I do think this opens the possibility of adding a content type that might not exist.